### PR TITLE
Use GitHub Actions to test Windows x86

### PR DESCRIPTION
**Pull request by aminya**
This is a patch for #12 
It removes Appveyor completely and uses GitHub actions for testing the Node x86 on Windows.

### CODE REVIEW

**Comment by aminya** [comment]
Despite installing Node x86 correctly, the Apm's bundled Node is still x64
https://github.com/atom-ide-community/apm/pull/28/checks?check_run_id=1073000962#step:5:319

This is because the process arch is still x64!
https://github.com/atom-ide-community/apm/blob/dd2a91c9f903ac7c4e0aa120c6b775311ca34636/script/download-node.js#L15

**Comment by DeeDeeG** [comment]
Maybe we can run a command to copy a 32-bit node.exe into `apm/bin/node.exe`... after the initial `npm install` completes.

Then `bin/npm rebuild` and `bin/npm test`

**Comment by DeeDeeG** [comment]
Or... maybe `script/bundled-node-version.js` should be changed: Maybe it really should follow the user's installed node arch, rather than the arch of the user's system?

**Comment by aminya** [comment]
No in a true x86 system this works without any problems: 
https://ci.appveyor.com/project/aminya/apm/builds/35033322/job/ry6yr3w02c9s0ene#L321

Unfortunately, GitHub Actions supports x64 agents.

**Comment by DeeDeeG** [comment]
I did a quick test. On 64-bit windows and 32-bit Node, `process.arch` returned 32-bit something, I think it was `ia32`.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -19,13 +19,20 @@ jobs:
           - 10
           - 12
           - 14
+        node_arch:
+          - x64
+        include:
+          - node: 12
```
Comment (DeeDeeG) [inline .github/workflows/CI.yml:25]: Typo? This is `node` whereas above it is `node_version`?

Comment (DeeDeeG) [reply .github/workflows/CI.yml:25]: This typo could cause the task to not actually install a new copy of Node, if I understand correctly.

 From: https://github.com/actions/setup-node#v2-beta

> The `node-version` input is optional.  If not supplied, node which is in your PATH will be used.

Comment (DeeDeeG) [reply .github/workflows/CI.yml:25]:
```suggestion
          - node_version: 12
```

**Comment by DeeDeeG** [comment]
I think it's cool that you patched the GitHub Actions task, by the way. Should make testing on Windows 32-bit easier. :+1:

File: .github/workflows/CI.yml (hunk 1 above)
Comment (DeeDeeG) [reply .github/workflows/CI.yml:25]: @aminya this PR works with the small typo fix I posted above.

https://github.com/DeeDeeG/apm/runs/1077130953#step:5:308

Comment (aminya) [reply .github/workflows/CI.yml]: I added a warning in https://github.com/actions/setup-node/pull/194/commits/2a5c060ce28fea5df265375fde1fa37b767a310b

**Comment by aminya** [comment]
This now uses my fork. I'll move to official releases once https://github.com/actions/setup-node/pull/194 is merged.

**Pull request merged**